amd_openvx_extensions/amd_rpp/source/tensor/Normalize.cpp

```diff
@@ -77,17 +88,48 @@ static vx_status VX_CALLBACK refreshNormalize(vx_node node, const vx_reference *
             dst_roi[i].xywhROI.roiHeight = src_roi[i].xywhROI.roiHeight;
         }
         data->pSrcRoi = static_cast<unsigned *>(data->pSrcDims);
-    } else {
+    } else if (numDims == 4) {  // For NHWC/NCHW layouts
+        RpptROI *src_roi = reinterpret_cast<RpptROI *>(roi_tensor_ptr);
+        RpptROI *dst_roi = reinterpret_cast<RpptROI *>(roi_tensor_ptr_dst);
+        for (unsigned i = 0; i < data->inputTensorDims[0]; i++) {
+            // rocAL ROI for image formats is stored in XYWH format. Transpose kernel needs ROI for all dims so adding the channel ROI here
+            unsigned index = i * roiDims * 2;
+            if (data->inputLayout == vxTensorLayout::VX_NHWC) {
+                data->pSrcDims[index + 0] = src_roi[i].xywhROI.xy.y;       // StartH
+                data->pSrcDims[index + 1] = src_roi[i].xywhROI.xy.x;       // StartW
+                data->pSrcDims[index + 2] = 0;                             // StartC
+                data->pSrcDims[index + 3] = src_roi[i].xywhROI.roiHeight;  // EndH
+                data->pSrcDims[index + 4] = src_roi[i].xywhROI.roiWidth;   // EndW
+                data->pSrcDims[index + 5] = data->inputTensorDims[3];      // EndC - assign channel value from input dims
+            } else if (data->inputLayout == vxTensorLayout::VX_NCHW) {
+                data->pSrcDims[index + 0] = 0;                             // StartC
+                data->pSrcDims[index + 1] = src_roi[i].xywhROI.xy.y;       // StartH
+                data->pSrcDims[index + 2] = src_roi[i].xywhROI.xy.x;       // StartW
+                data->pSrcDims[index + 3] = data->inputTensorDims[1];      // EndC - assign channel value from input dims
+                data->pSrcDims[index + 4] = src_roi[i].xywhROI.roiHeight;  // EndH
+                data->pSrcDims[index + 5] = src_roi[i].xywhROI.roiWidth;   // EndW
+            }
+            dst_roi[i].xywhROI = src_roi[i].xywhROI;
+        }
+        data->pSrcRoi = static_cast<unsigned *>(data->pSrcDims);
+    } else {  // For other layouts which already has generic ROI
         data->pSrcRoi = static_cast<unsigned *>(roi_tensor_ptr);
+        Rpp32u *dst_roi = static_cast<unsigned *>(roi_tensor_ptr_dst);
+        for (unsigned i = 0; i < data->inputTensorDims[0]; i++) {
+            unsigned index = i * roiDims * 2;
```

Comment:
What is *2 corresponds to?

Reply:
For NHWC/NCHW layouts, roiDims value is 3. We need to create a buffer for BS * 3 * 2 to hold start and end values for HWC values